databroker/broker.py

```diff
@@ -913,7 +912,7 @@ def wrapped_f(*args, **kwargs):
                 # doc will pass through unchanged; fs_doc may be modified to
                 # replace some values with references to filestore.
                 if external_writers:
-                    fs_doc = copy.deepcopy(doc)
+                    fs_doc = dict(doc)
```

Comment:
This might be a problem?
I went with a deepcopy because when `fs_doc` got mutated it mutated the the yielded doc.
Does this fix this?